active/Expander/Expander.md

```diff
@@ -0,0 +1,155 @@
+
+Throughout Windows, different expander controls are used by various apps and surfaces.
+There’s currently no consistent way to address this common UX pattern.
+This control is needed for situations where expanding (instead of overlaying) content is needed.
+An Expander control is motivated by its use in many app scenarios and supporting developers in migrating from
+[WPF](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/controls/expander-overview?view=netframeworkdesktop-4.8)
+and [Windows Community Toolkit](https://docs.microsoft.com/en-us/windows/communitytoolkit/controls/expander).  
+
+# API Pages
+
+## Expander class
+
+Represents the control that displays a header and a collapsible content area.
+
+> Spec note: the Expander class and ExpandDirection enum names match WPF.
+
+An Expander control is a UI component that provides a standard interaction for
+showing more content in a container that pushes adjacent content while expanding and collapsing.
+An Expander cannot be light dismissed and is independent of the contents inside it, including controls.
+Expander should be used when some content is only relevant some of the time (for example to
+read more information or access additional options for an item). 
+
+![A collapsed Expander that is expanded (pushing downwards) and then collapsed. The Header has the text "This is in the header" and the Content has the text "This is in the content". There is a TextBlock saying "This TextBlock is Below the Expander" that is pushed away](images/Expander_with_textblock.gif)
+
+### Examples
+
+#### Create an Expander
+
+![A collapsed Expander that is expanded and then collapsed. The Header has the text "This is in the header" and the Content has the text "This is in the content".](images/Expander.gif)
+
+XAML
+~~~~
+<muxc:Expander x:Name="DefaultExpander" 
+    Header="This text is in the header" 
+    Content="This is in the content"/>
+~~~~
+
+#### Put controls inside an Expander
+
+![A collapsed Expander. The Header has the text "This is in the header" and the Content has a button with the text "This ToggleButton is in the Content". The Expander is expanded, the button is toggled on and off, and the Expander is collapsed. ](images/Expander_togglebutton.gif)
+
+XAML
+~~~~
+<muxc:Expander x:Name="Expander2" Header="This is in the header"> 
+    <ToggleButton>This ToggleButton is in the Content</ToggleButton>
+</muxc:Expander>
+~~~~
+
+Updated pictures with final visuals to come.
+
+### Theme Resources
+
+You can modify the look of an Expander by specifying Xaml resources in your app.
+For more info, see the
+[lightweight styling guide](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/xaml-styles#lightweight-styling).
+
+| Name| Description | 
+| :---------- | :------- | 
+| ExpanderChevronMargin | Chevron margin thickness| 
+| ExpanderChevronGlyph | Chevron glyph| 
+| ExpanderChevronSize | Chevron size|
+| ExpanderPopinVerticalOffset | vertical offset for animation| 
```

Comment:
How do I style the background for the header and content separately? Which one does the `Background` property apply to? Similarly for font properties and `Foreground`.

Is this where `HeaderTemplate` enters the picture?

Or is this a case where you have to edit the template, and we expect most people to accept the default background and font?

Reply:
I think the HeaderTemplate can be used to style the header, but @alexhinz knows more about this than me.

Reply:
`Header`, `HeaderTemplate`, and `HeaderTemplateSelector` are functionally analogous to `Content`, `ContentTemplate`, and `ContentTemplateSelector`, and they apply to the area of Expander that's to the left of the chevron. So because they don't span the entire border-to-border (inclusive) area of the header, they're likely not a good fit for that kind of styling. They also would not be aware of any Expander states, like if it's expanded, if a dev would like to change colors based on them.

So devs could retemplate to style the header and background areas separately. Adding in additional lightweight styling properties would help avoid the need to retemplate.

Reply:
I think lightweight styling properties would be popular.

* To convey state. E.g. "Tuesday has slots available, Wednesday is full, and Thursday is oversubscribed."
* To color-code the Expanders. E.g., Errors in red, Warnings in yellow.
* Everybody wants their colors to match their brand.

Reply:
Re @MikeHillberg question: Following the latest Expander PR merge the Background property now does propagates to the Header (while the Foreground, Fontsize, and FontFamily propagate to the Content). As an example, the following code results in an Expander where the entire header area (including 'under' the chevron) is Cyan, with default text styling, but the content area has default background color with text that is pink and Times New Roman font. (If there's a way to add pictures to a PR thread let me know!)

<controls:Expander x:Name="Expander1" Header="This is in the header"  Content="This is in the content" FontSize ="25" FontFamily="Times New Roman" Foreground="Magenta" Background="Cyan"/>

Reply:
Discussion: You could set the background of the Header using the Background property, and the background of the Content within the content itself.

Recommend: Maybe show an example?